### PR TITLE
GH-3056: Add Micrometer Metrics to Runtime Graph

### LINKED ISSUE #3056: Add micrometer metrics to the integration graph nodes

### PR DESCRIPTION
**Pull request by garyrussell**
Resolves https://github.com/spring-projects/spring-integration/issues/3056

### CODE REVIEW

**Comment by garyrussell** [comment]
Do not merge.

Needs rework tomorrow - this currently requires a call to `rebuild()` to get the stats updated - we need to store the meters in the graph nodes and fetch the current meter values.

File: src/reference/asciidoc/graph.adoc [hunk 1]
```diff
@@ -12,64 +12,83 @@ A Spring Integration application with only the default components would expose a
 [source,json]
 ----
 {
-  "contentDescriptor": {
-    "providerVersion": "4.3.0.RELEASE",
-    "providerFormatVersion": 1.0,
-    "provider": "spring-integration",
-    "name": "myApplication"
+  "contentDescriptor" : {
+    "providerVersion" : "5.2.0.RELEASE",
```
Comment (artembilan) [inline src/reference/asciidoc/graph.adoc]: Let's make this based on the `{project-version}` placeholder!
Let's hope `subs="normal"` in the `[source,json]` definition helps!

File: src/reference/asciidoc/graph.adoc [hunk 2]
```diff
@@ -12,64 +12,83 @@ A Spring Integration application with only the default components would expose a
...
+    "from" : 2,
+    "to" : 3,
+    "type" : "input"
+  } ]
 }
 ----
 ====
 
+NOTE: Version 5.2 has deprecated the legacy metrics in favor of Micrometer meters as discussed <<./metrics.adoc#metrics-management,Metrics Management>>.
+While not shown above, the legacy metrics (under the `stats` child node) will continue to appear in the graph but with an extra child node `"deprecated" : "stats are deprecated in favor of sendTimers and receiveCounters"`.
```
Comment (artembilan) [inline src/reference/asciidoc/graph.adoc]: I heard that we always have to add come before `but`. No? :curious:

**Review by artembilan — CHANGES_REQUESTED** [review]
Just a couple concerns

**Pull request merged**